codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -468,6 +464,9 @@ public boolean write(int allowedBytes) {
                             size == bytesWritten && endOfStream, writePromise);
                 } while (size != bytesWritten && allowedBytes > bytesWritten);
                 size -= bytesWritten;
+                if (size == 0 && endOfStream) {
+                    lifecycleManager.closeLocalSide(stream, promise);
```

Comment:
That is fair.  My concern was how much needs to be exposed for extension frames.  Seemed like a nice encapsulation to have this cleanup functionality (which should be common to all frame types) shared in the base class.  I guess we can deal with this when extension frames are formally supported by the flow controller.